### PR TITLE
BS-123 | icd-10 extensions outside the Bahmni reports code

### PR DESCRIPTION
**Pull request by daviemukungi**
## Requirements

- [x] This PR has a proper title that briefly describes the work done
- [x] I have squashed / amended the comments to make it more readable
- [x] I have included link to all the JIRA ticket('s)
- [x] I wrote the code as a pair or atleast performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Summary
Moved icd10 - extensions Outside Bahmni-Reports which will be usued to build the ICD10 extension jar and loaded to reports container for execution.

## Screenshot
snomedScreenshot

## JIRA tickets
https://bahmni.atlassian.net/browse/BS-123

### CODE REVIEW

File: src/main/java/org/bahmni/reports/extensions/util/Templates.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                .setCrosstabGroupStyle(crosstabGroupStyle)
+                .setCrosstabGroupTotalStyle(crosstabGroupTotalStyle)
+                .setCrosstabGrandTotalStyle(crosstabGrandTotalStyle)
+                .setCrosstabCellStyle(crosstabCellStyle)
+                .ignorePagination()
+                .setTableOfContentsCustomizer(tableOfContentsCustomizer);
+
+        currencyType = new CurrencyType();
+
+        HyperLinkBuilder link = hyperLink("http://www.dynamicreports.org");
```
Comment (gsluthra) [inline src/main/java/org/bahmni/reports/extensions/util/Templates.java]: What is this URL for?

**Comment by gsluthra** [comment]
There don't seem to be any unit tests. Is that going to be handled as a different PR?

**Review by gsluthra — CHANGES_REQUESTED** [review]

**Comment by daviemukungi** [comment]
> There don't seem to be any unit tests. Is that going to be handled as a different PR?

@gsluthra, I have added all the required unit tests.

File: src/main/java/org/bahmni/reports/extensions/util/Templates.java (hunk 1 above)
Comment (daviemukungi) [reply src/main/java/org/bahmni/reports/extensions/util/Templates.java]: This URL was part of the existing code in the Templates class in bahmni reports which contains the reports ui styles and formats.  I have removed the class from this repository and created the styles manually in the icd10ResultSetExtension class where they are needed.  @gsluthra 

Below is the code snippet for the ui styles created.
```
private static class reportStyle {
        {
            columnStyle = stl.style().setPadding(2)
                    .setVerticalAlignment(VerticalAlignment.MIDDLE)
                    .setHorizontalAlignment(HorizontalAlignment.CENTER);
        }
    }
```

**Review by vijayanandtwks — APPROVED** [review]

**Review by gsluthra — APPROVED** [review]

**Pull request merged**